tests/parse_metadata.py

```diff
@@ -240,6 +254,20 @@ def read_metadata(distribution: str) -> StubMetadata:
     assert type(uploaded_to_pypi) is bool
     partial_stub: object = data.get("partial_stub", True)
     assert type(partial_stub) is bool
+    requires_python_str: object = data.get("requires_python")
+    oldest_supported_python = _get_oldest_supported_python()
+    oldest_supported_python_specifier = Specifier(f">={oldest_supported_python}")
+    if requires_python_str is None:
+        requires_python = oldest_supported_python_specifier
+    else:
+        assert isinstance(requires_python_str, str)
```

Comment:
I prefer to be extremely strict about the validation we do in this file: I don't know how a `str` subclass could turn up here, so something pretty scary is happening if it does. I'd like to know about that if it happens! (It won't. But the point stands.)

```suggestion
        assert type(requires_python_str) is str
```